static/app/views/discover/results.tsx

```diff
@@ -918,50 +920,59 @@ const TipContainer = styled(MarkedText)`
   }
 `;
 
-type SavedQueryState = DeprecatedAsyncComponent['state'] & {
-  savedQuery?: SavedQuery | null;
-};
+function SavedQueryAPI(props: Omit<Props, 'savedQuery' | 'loading' | 'setSavedQuery'>) {
+  const queryClient = useQueryClient();
+  const {organization, location} = props;
 
-class SavedQueryAPI extends DeprecatedAsyncComponent<Props, SavedQueryState> {
-  shouldReload = true;
+  const queryKey = useMemo(
+    (): ApiQueryKey => [
+      `/organizations/${organization.slug}/discover/saved/${location.query.id}/`,
+    ],
```

Comment:
does this page have pagination and stuff that needs to get passed here?

Reply:
I don't think so, this should be the single result that we're looking at instead of the list. Hence the `id` bit in this query.

`location` and all that stuff is getting passed down into `<Results {...props}` below.

Reply:
double checking, but the network traffic before/after looks the same:

![SCR-20250506-kiei](https://github.com/user-attachments/assets/da547af8-29a2-429c-967a-062e59d4041f)

Somewhere else ([not here](https://github.com/getsentry/sentry/blob/51e695431604d0bc7b29103e9262aa74836e14cf/static/app/actionCreators/discoverSavedQueries.tsx), but close) we make that request to get the list of things, and in this component we're fetching the individual item.